src/test/java/com/mathworks/ci/RunMatlabBuildBuilderTest.java

```diff
@@ -0,0 +1,335 @@
+        project.getBuildWrappersList().add(this.buildWrapper);
+        RunMatlabBuildBuilderTester tester =
+                new RunMatlabBuildBuilderTester(matlabExecutorAbsolutePath, "-positive");
+        scriptBuilder.setTasks("");
+        project.getBuildersList().add(tester);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        jenkins.assertBuildStatus(Result.SUCCESS, build);
+
+    }
+
+    /*
+     * Test to verify builder correctly sets tasks that user entered.
+     */
+    @Test
+    public void verifyBuildPicksTheCorrectBuildBatch() throws Exception {
+        this.buildWrapper.setMatlabBuildWrapperContent(new MatlabBuildWrapperContent(Message.getValue("matlab.custom.location"), getMatlabroot("R2018b")));
+        project.getBuildWrappersList().add(this.buildWrapper);
+        scriptBuilder.setTasks("compile");
+        project.getBuildersList().add(this.scriptBuilder);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        jenkins.assertLogContains("Generating MATLAB script with content", build);
+        jenkins.assertLogContains("buildtool compile", build);
+    }
+
+    /*
+     * Test to verify if MATLAB scratch file is not generated in workspace for this builder.
+     */
+    @Test
+    public void verifyMATLABscratchFileNotGenerated() throws Exception {
+        this.buildWrapper.setMatlabBuildWrapperContent(new MatlabBuildWrapperContent(Message.getValue("matlab.custom.location"), getMatlabroot("R2018b")));
+        project.getBuildWrappersList().add(this.buildWrapper);
+        scriptBuilder.setTasks("");
+        project.getBuildersList().add(this.scriptBuilder);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        File matlabRunner = new File(build.getWorkspace() + File.separator + "runMatlabTests.m");
+        Assert.assertFalse(matlabRunner.exists());
+    }
+    
+    /*
+     * Test to verify build supports resolving environment variable (For matrix builds).
+     */
+    @Test
+    public void verifyBuildSupportsEnvVar() throws Exception {
+        EnvironmentVariablesNodeProperty prop = new EnvironmentVariablesNodeProperty();
+        EnvVars var = prop.getEnvVars();
+        var.put("TASKS", "compile");
+        jenkins.jenkins.getGlobalNodeProperties().add(prop);
+        this.buildWrapper.setMatlabBuildWrapperContent(new MatlabBuildWrapperContent(Message.getValue("matlab.custom.location"), getMatlabroot("R2018b")));
+        project.getBuildWrappersList().add(this.buildWrapper);
+        scriptBuilder.setTasks("$TASKS");
+        project.getBuildersList().add(scriptBuilder);
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+        jenkins.assertLogContains("buildtool compile", build);
+    }
+    
+    /*
+     * Test to verify if appropriate MATLAB runner file is copied in workspace.
+     * 
+     * NOTE: This test assumes there is no MATLAB installed and is not on System Path.
```

Comment:
Would love to remove this requirement in the future (in all other tests as well). It took me a while to figure out why tests were failing when I first tried to build.

Reply:
the rationale behind  tests is  considering the environment of jenkins.  where it would run without actual MATLAB instances. Hence whatever we could test which is closer to actual would be good idea

Reply:
I also think we could restrict the test cases related to RunMatlabBuilderBuild Class here instead of re running duplicates.

Reply:
I am having trouble understanding which duplicate tests to remove. `verifyMATLABrunnerFileGenerated` and `verifyMatrixBuildFails`?